src/internal/ethers-event.ts

```diff
@@ -0,0 +1,322 @@
+import {
+  AlchemyEventFilter,
+  AlchemyEventType,
+  AlchemyMinedTransactionsAddress,
+  AlchemyMinedTransactionsEventFilter,
+  AlchemyPendingTransactionsEventFilter,
+  AlchemySubscription,
+  NonEmptyArray
+} from '../types/types';
+import {
+  ALCHEMY_EVENT_TYPES,
+  ALCHEMY_MINED_TRANSACTIONS_EVENT_TYPE,
+  ALCHEMY_PENDING_TRANSACTIONS_EVENT_TYPE,
+  Event
+} from './internal-types';
+
+/**
+ * Wrapper class around the ethers `Event` class in order to add support for
+ * Alchemy's custom subscriptions types.
+ *
+ * The serialization and deserialization mechanism requires the order of the
+ * fields to be the same across different {@link ALCHEMY_EVENT_TYPES}. Before
+ * using a getter on a new event filter, make sure that the position of the
+ * field in serialization is correct.
+ *
+ * The getters on this class deserialize the event tag generated by
+ * {@link getAlchemyEventTag} into the original fields passed into the event.
+ */
+export class EthersEvent extends Event {
```

Comment:
Pretty interesting! What you have here is fine since there's only a few event types so it's easy to keep track of all of them.

If we were to have a lot of event types, we'd probably want to define a "schema" for each event type that might look something like
```ts
const EVENT_TYPE_TO_SCHEMA = {
  [ALCHEMY_PENDING_TRANSACTIONS_EVENT_TYPE]:
    [
      ["fromAddress": StringParam],
      ["toAddress": StringParam],
      ["hashesOnly": BooleanParam],
    ],
  [ALCHEMY_MINED_TRANSACTIONS_EVENT_TYPE]:
    [
      ["addresses", AddressesParam],
      ["includeRemoved", BooleanParam],
      ["hashesOnly", BooleanParam],
    ],
}
```
where each of those `Param` objects contains the logic of how to serialize and deserialize a field of that type. Then you might use the above in a helper `deserialize` that acts like
```ts
const fields = deserialize(eventTag)
// returns e.g.:
//  {
//     "method": "alchemy_pendingTransactions",
//     "fromAddress": "0xABC",
//     "toAddress": ["0xDEF", "0xGHI"],
//     "hashesOnly: true
//   }
Object.assign(this, fields);
```
But that's a ton more work and we can worry about that if we end up with a bunch of event types later on.

Reply:
Thanks! Will keep this in mind as we add more endpoints.